lib/subscription-connection.js

```diff
@@ -187,6 +187,7 @@ module.exports = class SubscriptionConnection {
       {}, // rootValue
       {
         ...context,
+        __currentQuery: print(document),
```

Comment:
I think this would be too costly to do it all the time. Maybe we can do it as a getter instead?

Reply:
I didn't dig into subscription protocols, but as far as I can understand, there can be some cases when query arrives as a DocumentNode straight away. My initial thought was just to ignore this case and set __currentQuery when query is string - can it work?

Don't quite understood about getter - where?

Reply:
> I didn't dig into subscription protocols, but as far as I can understand, there can be some cases when query arrives as a DocumentNode straight away. My initial thought was just to ignore this case and set __currentQuery when query is string - can it work?

To be honest I'm not sure. Maybe add some tests.

> Didn't quite understood about getter - where?

Instead of calling `print(document)` when the object is created, add a getter to said object and call `print()` when it is accessed. In this way, if no one access `__currentQuery`, there will not overhead.

Reply:
Understood, thank you, I'll change it to a getter.

Reply:
This needs work!